test.js

```diff
@@ -216,6 +253,128 @@ test('sign and verify', function (t) {
   })
 })
 
+test('sign and verify with RSA and options', function (t) {
+  t.plan(2)
+
+  t.test('server methods', function (t) {
+    t.plan(2)
+
+    const fastify = Fastify()
+    fastify.register(jwt, { secret: { key: privateKey, passphrase: publicKey }, options: { issuer: 'Some issuer', subject: 'Some subject', audience: 'Some audience', algorithm: 'RS256' } })
+
+    fastify
+      .ready()
+      .then(function () {
+        t.test('synchronous', function (t) {
+          t.plan(1)
+
+          const token = fastify.jwt.sign({ foo: 'bar' })
+          const decoded = fastify.jwt.verify(token)
+
+          t.is(decoded.foo, 'bar')
+        })
+
+        t.test('with callbacks', function (t) {
+          t.plan(3)
+
+          fastify.jwt.sign({ foo: 'bar' }, function (error, token) {
+            t.error(error)
+
+            fastify.jwt.verify(token, function (error, decoded) {
+              t.error(error)
+              t.is(decoded.foo, 'bar')
+            })
+          })
+        })
+      })
+  })
+
+  t.test('route methods', function (t) {
+    t.plan(2)
+
+    const fastify = Fastify()
+    fastify.register(jwt, { secret: 'test' })
+
+    fastify.post('/signSync', function (request, reply) {
+      reply.jwtSign(request.body)
+        .then(function (token) {
+          return reply.send({ token })
+        })
```

Comment:
Can you change this to:

```js
     return reply.jwtSign(request.body)
        .then(function (token) {
          return { token }
        })
```

In this way errors do not go to 'unhanledRejection'